iaso/tests/api/test_group_sets.py

```diff
@@ -0,0 +1,373 @@
+from django.utils.timezone import now
+from rest_framework import status
+
+from iaso import models as m
+from iaso.models import GroupSet
+from iaso.test import APITestCase
+from hat.audit.models import Modification
+
+
+class GroupSetsAPITestCase(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.now = now()
+
+        cls.data_source = m.DataSource.objects.create(name="Default source")
+        cls.data_source_2 = m.DataSource.objects.create(name="Source 2")
+        cls.source_version_1 = m.SourceVersion.objects.create(data_source=cls.data_source, number=1)
+        cls.source_version_2 = m.SourceVersion.objects.create(data_source=cls.data_source_2, number=2)
+
+        star_wars = m.Account.objects.create(name="Star Wars", default_version=cls.source_version_1)
+        marvel = m.Account.objects.create(name="Marvel")
+
+        cls.acccount_1_user_1 = cls.create_user_with_profile(
+            username="yoda", account=star_wars, permissions=["iaso_org_units"]
+        )
+        cls.acccount_1_user_2 = cls.create_user_with_profile(username="chewbacca", account=star_wars)
+        cls.acccount_3_user_1 = cls.create_user_with_profile(
+            username="raccoon", account=marvel, permissions=["iaso_org_units"]
+        )
+
+        cls.project_1 = m.Project.objects.create(
+            name="Hydroponic gardens", app_id="stars.empire.agriculture.hydroponics", account=star_wars
+        )
+
+        cls.project_2 = m.Project.objects.create(
+            name="New Land Speeder concept", app_id="stars.empire.agriculture.land_speeder", account=star_wars
+        )
+
+        cls.src_1_group_1 = m.Group.objects.create(name="Councils", source_version=cls.source_version_1)
+        cls.src_1_group_2 = m.Group.objects.create(name="Hidden", source_version=cls.source_version_1)
+
+        cls.src_2_group_1 = m.Group.objects.create(name="Assemblies", source_version=cls.source_version_2)
+
+        cls.project_1.data_sources.add(cls.data_source)
+        cls.project_1.save()
+        cls.data_source.account = star_wars
+        cls.data_source.default_version = cls.source_version_1
+        cls.data_source.projects.add(cls.project_1)
+        cls.data_source.save()
+
+        cls.data_source_2.account = star_wars
+        cls.data_source_2.projects.add(cls.project_1)
+        cls.data_source_2.save()
+
+        cls.acccount_1_user_1.iaso_profile.projects.add(cls.project_1)
+
+    def test_authentification_required(self):
```

Comment:
I would maybe add:

- a test for retrieve/POST/PATCH/DELETE without authentication
- a test for list/retrieve/POST/PATCH/DELETE without `org_units` permission